schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AbstractAvroSerDesProtocolHandler.java

```diff
@@ -42,13 +42,13 @@
      */
     public static final String READER_SCHEMA = "reader.schema";
 
-    private final DefaultAvroSerDesHandler defaultAvroSerDesHandler;
+    private final AvroSerDesHandler avroSerDesHandler;
 
     protected final Byte protocolId;
 
-    protected AbstractAvroSerDesProtocolHandler(Byte protocolId) {
+    protected AbstractAvroSerDesProtocolHandler(Byte protocolId, AvroSerDesHandler avroSerDesHandler) {
```

Comment:
As noted though in MessageContextSerDesHandler you now extend the abstract class but don't delegate to it for payload serialisation (as previously was doing), as such don't benefit from ability to have the multi payload serialisation and have had to hard code it, this way we can keep this delegation as before.